Source/WebCore/platform/graphics/cg/ImageDecoderCG.cpp

```diff
@@ -68,10 +68,12 @@ static RetainPtr<CFMutableDictionaryRef> createImageSourceOptions()
     CFDictionarySetValue(options.get(), kCGImageSourceShouldPreferRGB32, kCFBooleanTrue);
     CFDictionarySetValue(options.get(), kCGImageSourceSkipMetadata, kCFBooleanTrue);
 
+#if !HAVE(CGIMAGESOURCE_DISABLE_HARDWARE_DECODING)
     if (ProcessCapabilities::isHardwareAcceleratedDecodingDisabled())
         CFDictionarySetValue(options.get(), kCGImageSourceUseHardwareAcceleration, kCFBooleanFalse);
+#endif
 
-#if HAVE(IMAGE_RESTRICTED_DECODING) && USE(APPLE_INTERNAL_SDK)
+#if HAVE(IMAGE_RESTRICTED_DECODING) && USE(APPLE_INTERNAL_SDK) && !HAVE(CGIMAGESOURCE_ENABLE_RESTRICTED_DECODING)
```

Comment:
I'm not sure about this part. Isn't "restricted decoding" about media decoders? Do those consult the global ImageIO flag?

Reply:
That is a good point. I will have to look closer into that.

Thanks for reviewing!